### PR TITLE
mohan08p mongodb cheat sheet

### PR DESCRIPTION
**Pull request by mohan08p**
What does your Instant Answer do?

Show all The MongoDB basic queries and keys as cheat sheet.

What is the data source for your Instant Answer? (Provide a link if possible)

https://blog.codecentric.de/files/2012/12/MongoDB-CheatSheet-v1_0.pdf

Why did you choose this data source?

One of the Best and even recommended database for open-source development.

Are there any other alternative (better) data sources?

There are chances.

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

Coder, Open source, Git users, Geeks, Hackers, etc.

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?

Yes, https://duck.co/ideas/idea/6304/mongodb-cheet-sheet

Are you having any problems? Do you need our help with anything?

No

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)
screenshot from 2015-08-30 11 35 06

IA Page: https://duck.co/ia/view/mongodb_cheat_sheet

### CODE REVIEW

**Comment by jagtalon** [comment]
@mohan08p having links as the basic information might not be that useful and it's kind of hard to use because it's not clickable. It might be better to assume that they already have MongoDB and jump right into the deets like https://duckduckgo.com/?q=photoshop+cheat+sheet&ia=cheatsheet

File: share/goodie/cheat_sheets/json/mongodb.json [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+                "val": "9"
+            },
+            {
+                "key": "Null",
+                "val": "10"
+            }
+        ],
+         "Inserting Documents": [
+            {
+                "key": "db.ships.insert({name:'USS Enterprise-D',operator:'Starfleet',type:'Explorer',class:'Galaxy',crew:750,codes:[10,11,12]})",
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/json/mongodb.json]: You need to escape the `{` and `}` as described in https://duck.co/duckduckhack/goodie_cheat_sheets

**Comment by mohan08p** [comment]
thanks @jagtalon : I had that thought, even I didn't assign hyperlink to them. Just removed those things. Thanks so much.

File: share/goodie/cheat_sheets/json/mongodb.json (hunk 1 above)
Comment (mohan08p) [inline share/goodie/cheat_sheets/json/mongodb.json]: I am not getting. Can you please, explain a bit more ?

Comment (mintsoft) [inline share/goodie/cheat_sheets/json/mongodb.json]: @mohan08p you need to replace `{` with `\\{` https://duck.co/duckduckhack/goodie_cheat_sheets#escaping-special-characters

File: share/goodie/cheat_sheets/json/mongodb.json [hunk 2]
```diff
@@ -0,0 +1,342 @@
+{
+    "id": "mongodb_cheat_sheet",
+    "name": "MongoDB",
+    "description": "Mongo shell",
+    "metadata": {
+        "sourceName": "MongoDB - Cheat Sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/mongodb.json]: Let's change this to `Codecentric Blog`

File: share/goodie/cheat_sheets/json/mongodb.json [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        "sourceUrl": "https://blog.codecentric.de/files/2012/12/MongoDB-CheatSheet-v1_0.pdf"
+    },
+    "section_order": [
+	    "Basic Conceptes & Shell Commands",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/mongodb.json]: Please fix your indentation here, these should all have a uniform indentation. We use 4-space, soft tabs

File: share/goodie/cheat_sheets/json/mongodb.json [hunk 4]
```diff
@@ -0,0 +1,342 @@
+{
+    "id": "mongodb_cheat_sheet",
+    "name": "MongoDB",
+    "description": "Mongo shell",
+    "metadata": {
+        "sourceName": "MongoDB - Cheat Sheet",
+        "sourceUrl": "https://blog.codecentric.de/files/2012/12/MongoDB-CheatSheet-v1_0.pdf"
+    },
+    "section_order": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/mongodb.json]: Let's use the `terminal` template for this Cheat Sheet: `"template_type": "terminal",`

**Comment by moollaza** [comment]
@mohan08p thanks a lot for contributing, and sorry for the delay! I've left you some feedback that needs to be implemented. Once things are updated please let me so so I can review this again. If you have any questions, feel free to ask them here!

**Comment by moollaza** [comment]
@mohan08p Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

I'm closing this one for now, but if you'd like to finish this up, we'll gladly open it back up for you!

**Comment by javathunderman** [comment]
@moollaza Working on this!

**Pull request closed without merging**